boot/bootutil/include/bootutil/fault_injection_hardening.h

```diff
@@ -214,19 +214,9 @@ fih_int fih_int_encode(int x)
 }
 
 /* Standard equality. If A == B then 1, else 0 */
-#define fih_eq(x, y) ((x == y) && fih_delay() && !(y != x))
-#define fih_not_eq(x, y) ((x != y) && fih_delay() && !(y == x))
-// __attribute__((always_inline)) inline
-// int fih_eq(fih_ret x, fih_ret y)
-// {
-//     return (x == y) && fih_delay() && !(x != y);
-// }
-
-// __attribute__((always_inline)) inline
-// int fih_not_eq(fih_ret x, fih_ret y)
-// {
-//     return (x != y) && fih_delay() && !(x == y);
-// }
+#define FIH_EQ(x, y) ((x == y) && fih_delay() && !(y != x))
+#define FIH_NOT_EQ(x, y) ((x != y) || !fih_delay() || !(y == x))
```

Comment:
Does it improve security? If do so then how?

Reply:
I have seen that using inlined functions lead to more complex (and vulnerable) code at assembly level than using macros.  .